### PR TITLE
fix: use types from atom-ide-base

### PR DESCRIPTION
**Pull request by aminya**
### Description
- This PR uses the types from atom-ide-base.
- This fixes the typescript errors when someone uses atom-languageclient

I compared the types in atom-languageclient with the types we had here in atom-ide-base, and I added the missing ones into atom-ide-base. 
See https://github.com/atom-community/atom-ide-base/pull/57

We no longer need to maintain the types in two places.

### CODE REVIEW

**Comment by aminya** [comment]
The error here is an internal error of atom-ide-base because JSX is defined by React (or Etch). Here React is not used, so the code throws an error. I think the solution is to use `React.JSX.Element` in the types. 

https://github.com/atom-community/atom-ide-base/blob/8e952b465a6485eee76881cc201108e5bd853668/types-packages/datatip.d.ts#L72

File: lib/auto-languageclient.ts [hunk 1]
```diff
@@ -513,7 +514,7 @@ export default class AutoLanguageClient {
   protected onDidInsertSuggestion(_arg: ac.SuggestionInsertedEvent): void { }
 
   // Definitions via LS documentHighlight and gotoDefinition------------
-  public provideDefinitions(): atomIde.DefinitionProvider {
+  public provideDefinitions(): DefinitionProvider {
```
Comment (aminya) [inline lib/auto-languageclient.ts:517]: `wordRegExp` is missing here in atom-languageclient

File: test/adapters/datatip-adapter.test.ts [hunk 2]
```diff
@@ -50,14 +50,17 @@ describe('DatatipAdapter', () => {
...
-        invariant(snippet.type === 'snippet');
-        expect((snippet as any).grammar.scopeName).equal('text.plain.null-grammar');
-        expect(snippet.value).equal('test snippet');
+          const snippet = datatip.markedStrings[1];
+          expect(snippet.type).equal('snippet');
+          invariant(snippet.type === 'snippet');
+          expect((snippet as any).grammar.scopeName).equal('text.plain.null-grammar');
+          expect(snippet.value).equal('test snippet');
+        }
+        // else the React variant
```
Comment (aminya) [inline test/adapters/datatip-adapter.test.ts:63]: atom-languageclient only considered the markedString version as Datatip type

File: lib/adapters/notifications-adapter.ts [hunk 3]
```diff
@@ -10,9 +10,10 @@ import {
   NotificationOptions,
   NotificationExt,
 } from 'atom';
-import {
-  NotificationButton,
-} from 'atom-ide';
+
+export interface NotificationButton {
```
Comment (aminya) [inline lib/adapters/notifications-adapter.ts:14]: NotificationButton is not an ide type so I moved it here

File: lib/auto-languageclient.ts [hunk 4]
```diff
@@ -2,7 +2,7 @@ import * as cp from 'child_process';
 import * as ls from './languageclient';
 import * as rpc from 'vscode-jsonrpc';
 import * as path from 'path';
-import * as atomIde from 'atom-ide';
+import type { OutlineProvider, Outline, DefinitionProvider, DefinitionQueryResult, FindReferencesProvider, FindReferencesReturn, Datatip, DatatipService, TextEdit, RangeCodeFormatProvider, FileCodeFormatProvider, OnSaveCodeFormatProvider, OnTypeCodeFormatProvider, CodeAction, CodeHighlightProvider, CodeActionProvider, Diagnostic, RefactorProvider, BusySignalService, SignatureHelpRegistry, ConsoleService } from 'atom-ide-base';
```
Comment (aminya) [inline lib/auto-languageclient.ts:5]: #115 Will take care of formatting later

**Review by UziTech — APPROVED** [review]

**Pull request merged**